server/guides/guideController.js

```diff
@@ -26,20 +26,45 @@ module.exports = {
     });
   },
 
-  // findGuide: function(req, res, next, guideId){
-
-  // },
-
-  // showGuide: function(req, res, next){
-
-  // },
+  findGuide: function(req, res, next, guideId) {
+    Guide.findOne({ _id: guideId }, function(err, guide) {
+      if (guide) {
+        req.guide = guide;
+        next();
+      } else if (err) {
+        next(err);
+      } else {
+        next(new Error("No Guide with that Id"));
+      }
+    });
+  },
 
-  // editGuide: function(req, res, next){
+  showGuide: function(req, res) {
+    res.json(req.guide);
+  },
 
-  // },
+  editGuide: function(req, res, next) {
+    var errMes = "User not authorized to edit this Guide";
+    module.exports.isGuideCreator(req, next, errMes, function() {
+      var newGuide = { title: req.body.title, updatedAt: Date.now() };
+      var id = req.guide._id;
+      Guide.findByIdAndUpdate(id, newGuide, function(err, updatedGuide) {
+        err ? next(err) : res.send(updatedGuide);
+      });
+    });
+  },
 
-  // deleteGuide: function(req, res, next){
+  deleteGuide: function(req, res, next) {
+    var errMes = "User not authorized to delete this Guide";
+    module.exports.isGuideCreator(req, next, errMes, function() {
+      Guide.findByIdAndRemove(req.guide._id, function(err, result) {
+        result ? res.status(204).send() : next(new Error("No Guide with that Id"));
+      });
+    });
+  },
 
-  // }
+  isGuideCreator: function(req, next, errMes, cb) {
+    ''+req.user._id === ''+req.guide.userId ? cb() : next(new Error(errMes));
```

Comment:
@d4ncer I refactored this into the express middleware pattern. There was a merge conflict that I might have to rebase later to clean up the commit history, but the code looks much better now.